File: src/main/java/wav/demon/StatCraft/Magic/FishCode.java
package wav.demon.StatCraft.Magic;

public enum FishCode {

    FISH((byte)0),
    TREASURE((byte)1),
    JUNK((byte)2);

    private byte code;

    FishCode(byte code) {
        this.code = code;
    }

    public byte getCode() {
        return code;
    }

    public static FishCode fromCode(byte code) {
        for (FishCode fishCode : values()) {
            if (code == fishCode.getCode())
                return fishCode;
        }
        return null;
    }
}

File: src/main/kotlin/com/demonwav/statcraft/magic/FishCode.kt
/*
 * StatCraft Bukkit Plugin
 *
 * Copyright (c) 2016 Kyle Wood (DemonWav)
 * https://www.demonwav.com
 *
 * MIT License
 */

package com.demonwav.statcraft.magic

enum class FishCode(val code: Byte) {

    FISH    (0.toByte()),
    TREASURE(1.toByte()),
    JUNK    (2.toByte());

    companion object {
        @JvmStatic
        fun fromCode(code: Byte): FishCode? {
            for (fishCode in values()) {
                if (code == fishCode.code)
                    return fishCode
            }
            return null
        }
    }
}
